cleanlab/multilabel_classification/dataset.py

```diff
@@ -0,0 +1,369 @@
+
+
+    class_names : Iterable[str], optional
+        A list or other iterable of the string class names. The list should be in the order that
+        matches the label indices. So if class 0 is 'dog' and class 1 is 'cat', then
+        ``class_names = ['dog', 'cat']``.
+
+    confident_joint : np.ndarray, optional
+      An array of shape ``(K, 2, 2)`` representing a one-vs-rest formatted confident joint for multi-label data,
+      as returned by :py:func:`count.compute_confident_joint <cleanlab.count.compute_confident_joint>`.
+      Entry ``(c, i, j)`` in this array is the number of examples confidently counted into a ``(class c, noisy label=i, true label=j)`` bin,
+      where `i, j` are either 0 or 1 to denote whether this example belongs to class `c` or not
+      (recall examples can belong to multiple classes in multi-label classification).
+
+
+    Returns
+    -------
+    common_multilabel_issues : pd.DataFrame
+        DataFrame where each row corresponds to a Class (specified as the row-index) with columns "In Given Label", "In Suggested Label", "Num Examples", "Issue Probability".
+
+        * *In Given Label*: specifies whether the Class is True/False in the given label
+        * *Class*: If class_names is provided, the 'Class' column of the DataFrame will have the class name,
+            otherwise, the values will represent the class index.
+        * *In Suggested Label*: specifies whether the Class is  True/False in the suggested label (based on model prediction)
+        * *Num Examples*: Estimated number of examples with a label issue where this Class is True/False as specified "In Given Label" but cleanlab suggests it should be as specified In Suggested Label. I.e. the number of examples in your dataset where the Class was labeled as True but likely should have been False (or vice versa).
+        * *Issue Probability*: This is the  *Num Examples* column divided by the total number of examples in the dataset. It corresponds to the relative overall frequency of each type of label issue in your dataset.
+
+        By default, the rows in this DataFrame are ordered by "Issue Probability" (descending).
+    """
+
+    num_examples = _get_num_examples_multilabel(labels=labels, confident_joint=confident_joint)
+    summary_issue_counts = defaultdict(list)
+    if class_names is None:
+        num_classes = get_num_classes(labels=labels, pred_probs=pred_probs, multi_label=True)
+        class_names = list(range(num_classes))
+    if confident_joint is None:
+        y_one, num_classes = get_onehot_num_classes(labels, pred_probs)
+        label_issues_list, labels_list, pred_probs_list = _find_multilabel_issues_per_class(
+            labels=labels,
+            pred_probs=pred_probs,
+            confident_joint=confident_joint,
+            return_indices_ranked_by="self_confidence",
+        )
```

Comment:
```suggestion
      y_one, num_classes = get_onehot_num_classes(labels, pred_probs)
      label_issues_list, labels_list, pred_probs_list = _find_multilabel_issues_per_class(
          labels=labels,
          pred_probs=pred_probs,
          confident_joint=confident_joint,
          return_indices_ranked_by="self_confidence",
      )
```

Reply:
Just do the above always (regardless if `confident_joint` was provided or not). 
So `confident_joint` will be utilized inside of `_find_multilabel_issues_per_class` if it was provided.

The you can get rid of the second extra loop you have below specifically for confident_joint being provided.